tests/test_depsolver.py

```diff
@@ -0,0 +1,320 @@
+import pytest
+from pubtools.pulplib import YumRepository, RpmUnit, RpmDependency, ModulemdUnit
+from ubi_manifest.worker.tasks.depsolver.rpm_depsolver import (
+    Depsolver,
+    BATCH_SIZE_RESOLVER,
+)
+from ubi_manifest.worker.tasks.depsolver.models import UbiRepository
+
+
+def get_test_yum_repository(**kwargs):
+    pulp = kwargs.pop("pulp")
+    repo = YumRepository(**kwargs)
+    repo.__dict__["_client"] = pulp.client
```

Comment:
Hmm, I don't think code outside of pubtools-pulplib should be doing that.  Neither accessing `__dict__` nor writing to `_client` is meant as public API.  There is in general no supported way to attach an existing repository object to a client.

Shouldn't it use the FakeController (fake_pulp fixture)?  It has an insert_repository method.  The idea is that you can do this:

```python
# put a repo into the fake
ctrl.insert(YumRepository(id='my-repo', ...))

# now the regular client API works to get the same repo back out
client.get_repository('my-repo')
```

Actually it seems like you even use insert_repository later on, so I think you could already drop this assignment to _client and just make sure to get the repo back out of the client in each test before using it.